tests/test_extraction/test_text_extractor.py

```diff
@@ -0,0 +1,126 @@
+"""This script is for unit testing of text_extractor"""
```

Comment:
Might be a good idea to add tests for verifying that each method/function is working as intended. This will make it easier to debug issues in the future.

Reply:
Updated with tests for methods in string_utils.